python/semantic_kernel/planning/basic_planner.py

```diff
@@ -0,0 +1,208 @@
+        ]
+    }
+
+[AVAILABLE FUNCTIONS]
+{{$available_functions}}
+
+[GOAL]
+{{$goal}}
+
+[OUTPUT]
+"""
+
+
+class BasicPlanner:
+    """
+    Basic JSON-based planner for the Semantic Kernel.
+    """
+
+    def _create_available_functions_string(self, kernel: Kernel) -> str:
+        """
+        Given an instance of the Kernel, create the [AVAILABLE FUNCTIONS]
+        string for the prompt.
+        """
+        # Get a dictionary of skill names to all native and semantic functions
+        native_functions = kernel.skills.get_functions_view()._native_functions
+        semantic_functions = kernel.skills.get_functions_view()._semantic_functions
+        native_functions.update(semantic_functions)
+
+        # Create a mapping between all function names and their descriptions
+        # and also a mapping between function names and their parameters
+        all_functions = native_functions
+        skill_names = list(all_functions.keys())
+        all_functions_descriptions_dict = {}
+        all_functions_params_dict = {}
+
+        for skill_name in skill_names:
+            for func in all_functions[skill_name]:
+                key = skill_name + "." + func.name
+                all_functions_descriptions_dict[key] = func.description
+                all_functions_params_dict[key] = func.parameters
+
+        # Create the [AVAILABLE FUNCTIONS] section of the prompt
+        available_functions_string = ""
+        for name in list(all_functions_descriptions_dict.keys()):
+            available_functions_string += name + "\n"
+            description = all_functions_descriptions_dict[name]
+            available_functions_string += "description: " + description + "\n"
+            available_functions_string += "args:\n"
+
+            # Add the parameters for each function
+            parameters = all_functions_params_dict[name]
+            for param in parameters:
+                if not param.description:
+                    param_description = ""
+                else:
+                    param_description = param.description
+                available_functions_string += (
+                    "- " + param.name + ": " + param_description + "\n"
+                )
+            available_functions_string += "\n"
+
+        return available_functions_string
+
+    async def create_plan_async(
+        self,
+        goal: str,
+        kernel: Kernel,
+        prompt: str = PROMPT,
+    ) -> Plan:
+        """
+        Creates a plan for the given goal based off the functions that
+        are available in the kernel.
+        """
+
+        # Create the semantic function for the planner with the given prompt
+        planner = kernel.create_semantic_function(
+            prompt, max_tokens=1000, temperature=0.8
+        )
+
+        available_functions_string = self._create_available_functions_string(kernel)
+
+        # Create the context for the planner
+        context = ContextVariables()
```

Comment:
nit: it would be cleaner if the plan object could provide the ContextVariables() instead of having to put them together manually. 

If you don't want to address here, we could perhaps add it to the backlog?

Reply:
Let's clean it up in a future PR!